hydromt/data_adapter.py

```diff
@@ -602,48 +607,52 @@ def _parse_data_dict(data_dict, root=None, category=None):
         "GeoDataFrame": GeoDataFrameAdapter,
         "GeoDataset": GeoDatasetAdapter,
     }
-    # set data_type sections as source entry
-    # for backwards compatability
-    sources = copy.deepcopy(data_dict)
+    # NOTE: shouldn't the kwarg overwrite the dict/yml ?
     if root is None:
-        root = sources.pop("root", None)
-    for key in data_dict:
-        _key = key.replace("Adapter", "")
-        if _key in ADAPTERS:
-            _sources = data_dict.pop(key)
-            for name, source in _sources.items():
-                source["data_type"] = _key
-                sources[name] = source
+        root = data_dict.pop("root", None)
 
     # parse data
     data = dict()
-    for name, source in sources.items():
+    for name, source in data_dict.items():
+        source = source.copy()  # important as we modify with pop
         if "alias" in source:
             alias = source.pop("alias")
-            if alias not in sources:
-                raise ValueError(f"alias {alias} not found in sources.")
+            if alias not in data_dict:
+                raise ValueError(f"alias {alias} not found in data_dict.")
             # use alias source but overwrite any attributes with original source
             source_org = source.copy()
-            source = sources[alias].copy()
+            source = data_dict[alias].copy()
             source.update(source_org)
         if "path" not in source:
             raise ValueError(f"{name}: Missing required path argument.")
         data_type = source.pop("data_type", None)
         if data_type is None:
             raise ValueError(f"{name}: Data type missing.")
         elif data_type not in ADAPTERS:
-            raise ValueError(f"{name}: Data type {data_type} unkonwn")
+            raise ValueError(f"{name}: Data type {data_type} unknown")
         adapter = ADAPTERS.get(data_type)
         path = abs_path(root, source.pop("path"))
         meta = source.pop("meta", {})
         if "category" not in meta and category is not None:
             meta.update(category=category)
         # lower kwargs for backwards compatability
+        # FIXME this could be problamatic if driver kwargs conflict DataAdapter arguments
         source.update(**source.pop("kwargs", {}))
         for opt in source:
             if "fn" in opt:  # get absolute paths for file names
                 source.update({opt: abs_path(root, source[opt])})
-        data[name] = adapter(path=path, meta=meta, **source)
+        if "placeholders" in source:
+            options = source["placeholders"]
+            for combination in itertools.product(*options.values()):
```

Comment:
@laurenebouaziz I've shortened this part a bit, but it should still function like  you intended it to function. Could you double check with your data?

Reply:
thanks @DirkEilander, looks better indeed and still works the same for my test case!